TabnineSelfHosted/src/main/java/com/tabnineSelfHosted/binary/lifecycle/UserInfoService.kt

```diff
@@ -0,0 +1,38 @@
+package com.tabnineSelfHosted.binary.lifecycle
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.util.concurrency.AppExecutorUtil
+import com.intellij.util.messages.MessageBus
+import com.tabnineCommon.general.DependencyContainer
+import com.tabnineSelfHosted.binary.requests.userInfo.UserInfoRequest
+import com.tabnineSelfHosted.binary.requests.userInfo.UserInfoResponse
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+
+class UserInfoService {
+    private val scheduler = AppExecutorUtil.getAppScheduledExecutorService()
+    private val messageBus: MessageBus = ApplicationManager.getApplication().messageBus
+    private val updateLoopStarted = AtomicBoolean(false)
+    private val binaryRequestFacade = DependencyContainer.instanceOfBinaryRequestFacade()
+    private var userInfoResponse: UserInfoResponse? = null
```

Comment:
why do you keep the user info response here if its already saved in `UserInfoService`?